### PR TITLE
Strings2factor

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR aims to close https://github.com/tidymodels/recipes/issues/989 by adding some words of caution to the `step_string2factor()` documentation.

### CODE REVIEW

File: R/string2factor.R [hunk 1]
```diff
@@ -12,7 +12,22 @@
 #' @template step-return
 #' @family dummy variable and encoding steps
 #' @export
-#' @details If `levels` is given, `step_string2factor` will
+#' @details
+#'  In most cases, if you are planning to use `step_string2factor()`
+#'  without setting `levels`, you will be better off by converting
```
Comment (juliasilge) [inline R/string2factor.R:17]:
```suggestion
#'  without setting `levels`, you will be better off converting
```

File: R/string2factor.R [hunk 2]
```diff
@@ -12,7 +12,22 @@
 #' @template step-return
 #' @family dummy variable and encoding steps
 #' @export
-#' @details If `levels` is given, `step_string2factor` will
+#' @details
+#'  In most cases, if you are planning to use `step_string2factor()`
```
Comment (topepo) [inline R/string2factor.R:16]: Can you add a top-level statement that tells people to only use this in special cases and, instead, set the levels before using any tidymodels functions? That's the message and it is buried in a lot of minutiae.

Comment (juliasilge) [reply R/string2factor.R:16]: Do you mean in the description? That seems like a good addition to me to what is already in the details.

File: R/string2factor.R [hunk 3]
```diff
@@ -1,7 +1,9 @@
 #' Convert Strings to Factors
 #'
 #' `step_string2factor` will convert one or more character
-#'  vectors to factors (ordered or unordered).
+#'  vectors to factors (ordered or unordered). This step should only be
+#'  used in special cases. Factors should generally be created before using
+#'  any tidymodels functions.
```
Comment (juliasilge) [inline R/string2factor.R:6]:
```suggestion
#'  vectors to factors (ordered or unordered). Use this step only 
#'  in special cases (see Details) and instead convert strings to 
#'  factors before using any tidymodels functions.
```

**Pull request merged**